ui/pages/swaps/swaps.util.js

```diff
@@ -420,11 +420,12 @@ export async function fetchAggregatorMetadata(chainId) {
 
 export async function fetchTopAssets(chainId) {
   const topAssetsUrl = getBaseApi('topAssets', chainId);
-  const response = await fetchWithCache(
-    topAssetsUrl,
-    { method: 'GET', headers: clientIdHeader },
-    { cacheRefreshTime: CACHE_REFRESH_FIVE_MINUTES },
-  );
+  const response =
+    (await fetchWithCache(
+      topAssetsUrl,
+      { method: 'GET', headers: clientIdHeader },
+      { cacheRefreshTime: CACHE_REFRESH_FIVE_MINUTES },
+    )) || [];
```

Comment:
What if we update `fetchWithCache` to return an empty array instead? I'm seeing another instance that would break if this request returns null or undefined. 

`metamask-extension/ui/pages/confirmation/templates/add-ethereum-chain.js`
```
async function getAlerts(pendingApproval) {
  const alerts = [];
  const safeChainsList = await fetchWithCache(
    'https://chainid.network/chains.json',
  );
  const matchedChain = safeChainsList.find(
    (chain) =>
      chain.chainId === parseInt(pendingApproval.requestData.chainId, 16),
  );
```

Then in `metamask-extension/ui/pages/settings/networks-tab/networks-form/networks-form.js`
we could remove the `!Array.isArray(safeChainsList)` check in because the warning message for an empty list is called below this. 
```
      if (providerError || !Array.isArray(safeChainsList)) {
```

Reply:
Thanks for finding other places where it can be an issue and for the suggestion! I've looked into it and I don't think we can return an empty array from `fetchWithCache` if `responseJson` is null or undefined, because sometimes a calling function expects an object (maybe even a string or a number from some API calls), not an array, so that wouldn't be semantically right.

I've added a default empty array inside the `getAlerts` function and `validateTickerSymbol` function. I've also removed this code, as you suggested: ` || !Array.isArray(safeChainsList)`

Reply:
Awesome, the updates look great to me! I am still wondering about what you wrote:

> because sometimes a calling function expects an object (maybe even a string or a number from some API calls), not an array, so that wouldn't be semantically right.

My question is what function are you referring to exactly? I'm confused about how the function excepts an object that could be a string or number, but not an array given arrays are also objects.

Reply:
Hi, we synced up internally and are on the same page now. Here is what we discussed: If we would by default return an empty array for all possible API calls if they return null or undefined and the functions that are calling `fetchWithCache` can expect an API response to be either an array (special type of object), but also an object, string, number, etc., it can lead to the following issues:
```
const response = []; // Default response with an empty array.
console.log(response.split(","));
console.log(response.toFixed(0))
console.log(response.nativeCurrency.symbol);

console.log(response.split(",")); // returns Uncaught TypeError: response.split is not a function
console.log(response.toFixed(0)) // returns Uncaught TypeError: response.toFixed is not a function
console.log(response.nativeCurrency.symbol); // returns t read properties of undefined (reading 'symbol')
```

Reply:
Thank you @dan437! 🙏